### DISCUSSION TITLE
Getting error 4058 when installing via npm.

### DISCUSSION THREAD

**Discussion by mauroofcol** [tailwindlabs/tailwindcss#1862] [Help]
Good morning, 
Hope you all are having a good day,
I cloned the repo from github, when I run the install command it starts working and after a while it shows the error 4058, in which states that it couldn't find the file cli.js. I already tried to kill all node processes and work without vs code running as it was suggested in other similar situation but it didn't worked. Also I tried looked for the file manually but I couldn't find it. 

I would really appreciate any help-

**Comment by estevanmaito** [comment]
Looks like you already tried all of this: https://stackoverflow.com/a/49620827/4480553

Could you share the entire error message, your node version (`node -v`) and operating system?

**Reply by mauroofcol** [reply]
Thanks a lot for your help, yes! I tried that advise, this is the message I receive:
 
npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path C:\Users\User\Downloads\digital tools\programming\web development\css.engine\node_modules\tailwindcss\lib\cli.js
npm ERR! errno -4058
npm ERR! enoent ENOENT: no such file or directory, chmod 'C:\Users\User\Downloads\digital tools\programming\web development\css.engine\node_modules\tailwindcss\lib\cli.js'
npm ERR! This is related to npm not being able to find a file.
npm ERR! enoent

Node version: v12.17.0

OS: Windows 10 Education x64

**Reply by estevanmaito** [reply]
Try this first: rename your folder `css.engine` to something like `css-engine` and install again. Windows complains with special characters in folder names.

If it doesn't work, keep the changed name (`css-engine`) and try:

1. Delete `package-lock.json`
2. Delete `node_modules`
3. Run `npm cache clear --force`
4. Try `npm install` again

**Reply by mauroofcol** [reply]
done, it did not worked, I tried with the playground option but it didn't work either. Feels like something is missing, I am really new to this entire thing, I started with node and unix command line recently and utilized Hyper as a unix shell, maybe I did something wrong or I overlooked something, I don't know. I just want to try vanilla tailwind, it looks great.

**Reply by estevanmaito** [reply]
What terminal are you using? Git Bash, PowerShell, cmd...

**Reply by estevanmaito** [reply]
There are so many things that could go wrong developing on Windows (I use Windows too most of the time).

I would suggest you to change your terminal and use Git Bash instead. Here is a great tutorial to get it working https://www.stanleyulili.com/git/how-to-install-git-bash-on-windows/

Looks like a lot of work for now, but it will for sure save you time debugging things like this in the future.

**Reply by mauroofcol** [reply]
I just went to clean the kitchen I am really stressed lol, I am already using git bash, In the future I know I have to learn Linux, by now I am stuck with windows and its marvelous interface XD. If you have any idea about what else I can try I would really thank you, I will not have peace of mind until I have finished a project using tailwind.

**Reply by estevanmaito** [reply]
Wait.

I thought you were trying to install this REPOSITORY, I mean, the Tailwind CSS project, to develop in you computer.

> I cloned the repo from github

Are you actually trying to just use Tailwind CSS in a project to add style? Because if this is the case, you cleaned your kitchen for nothing 😂

You should be following these steps: https://tailwindcss.com/docs/installation

The steps starting with an empty folder:

1. In your terminal, from inside your project folder, run `npm init -y` to create a `package.json` file
2. Run `npm install tailwindcss`
3. Follow the instruction in the page above

**Reply by mauroofcol** [reply]
Lol, is in the second step you showed, running "npm install tailwindcss" where I get that error message. Now I understand that getting the repository was unnecessary XD, but the issue persists.

**Reply by estevanmaito** [reply]
It happened to me today, the same error!

I had a normal root project with `package.json` and `node_modules`, but I created another project inside this same project, so I had something like this:

```
app
├── node_modules
├── examples
│   └── simple
│       └──  package.json 👈
│
└── package.json
```

If I tried to `npm install` from inside `examples/simple` I would get the 4058 error. The only way to solve it was to close VSCode, remove `node_modules` from `examples/simple` and run `npm install` directly from the terminal/bash (inside `examples/simple`).

The problem, in my case, was that VSCode kept cached the `package-lock.json` from the root directory and it caused conflicts.

See if it can help you, try creating a new folder and starting from zero. Make sure you don't have another `package.json` up in you folders tree.

**Reply by mauroofcol** [reply]
I truly mean this: Thanks a LOT!

O Senhor te abençoe e te guarde;  o Senhor faça resplandecer o seu rosto sobre ti, e tenha misericórdia de ti;  o Senhor levante sobre ti o seu rosto, e te dê a paz.

**Reply by estevanmaito** [reply]
Obrigado, o mesmo pra vc irmão ;)

**Reply by vivekmisal8** [reply]
hi i got same error problem plz help
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! today4@0.1.0 start: `react-scripts start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the today4@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:

![Screenshot (6)](https://user-images.githubusercontent.com/83937990/131385198-def04100-7372-4e48-80bf-30f71b30f772.png)

i tried
npm update
npm install
there is no duplicate file of package-lock.json

**Comment by mauroofcol** [comment]
Good Morning Sr.
First, what do you want to do? what are you doing? which npm package are you installing or running? the error says is related to a react script, so why reach me for tailwind?, in my GitHub repo I have an unminified old version of tailwind, it is just a portion of the actual package, so it is hard for me to understand what is going on and how it is related to me. Now the image you send is a screenshot of a folder in your pc, that is not a log of an error. 
If you are using tailwind and react together and you have issues at this level please use raw css, tailwind will make your life harder reducing the readability of your code and summing layout bugs to logic bugs in the same files. If you are starting in tailwind, make some projects in plain HTML, CSS, and JS, then when you know how to use the library correctly and efficiently integrate it with react, I think there is a dedicated npm package for tailwind + react but I am not sure.
wish you a good day,



    En lunes, 30 de agosto de 2021 13:15:09 GMT-5, vivekmisal8 ***@***.***> escribió:  
 
 


hi i got same error problem plz help
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ***@***.*** start: react-scripts start
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the ***@***.*** start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:



i tried
npm update
npm install
there is no duplicate file of package-lock.json

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Bader-Idris** [comment]
when I had the same issue, it was simply because you have to access your repo in terminal by typing: cd directory\location
